build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -1416,6 +1416,129 @@ export const handlers: Handler[] = [
 				}
 			});
 
+			return result;
+		},
+	},
+	{
+		/**
+		 * Handler for {@link PolicyConfig.publicPackageRequirements}
+		 */
+		name: "npm-public-package-requirements",
+		match,
+		handler: async (packageJsonFilePath, rootDirectoryPath) => {
+			let packageJson: PackageJson;
+			try {
+				packageJson = JSON.parse(readFile(packageJsonFilePath));
+			} catch (err) {
+				return "Error parsing JSON file: " + packageJsonFilePath;
+			}
+
+			if (packageJson.private) {
+				// If the package is private, we have nothing to validate.
+				return;
+			}
+
+			const requirements =
+				getFluidBuildConfig(rootDirectoryPath).policy?.publicPackageRequirements;
+			if (requirements === undefined) {
+				// If no requirements have been specified, we have nothing to validate.
+				return;
+			}
+
+			const errors: string[] = [];
+
+			// Ensure the package has all required dev dependencies specified in the config.
+			if (requirements.requiredDevDependencies !== undefined) {
+				const devDependencies = Object.keys(packageJson.devDependencies ?? {});
+				for (const requiredDevDependency of requirements.requiredDevDependencies) {
+					if (!devDependencies.includes(requiredDevDependency)) {
+						errors.push(`Missing dev dependency: "${requiredDevDependency}"`);
+					}
+				}
+			}
+
+			// Ensure the package has all required scripts specified in the config.
+			if (requirements.requiredScripts !== undefined) {
+				const scriptNames = Object.keys(packageJson.scripts ?? {});
+				for (const requiredScript of requirements.requiredScripts) {
+					if (!scriptNames.includes(requiredScript.name)) {
+						// Enforce the script is present
+						errors.push(`Missing script: "${requiredScript.name}"`);
+					} else if (
+						requiredScript.bodyMustMatch === true &&
+						packageJson.scripts[requiredScript.name] !== requiredScript.body
+					) {
+						// Enforce that script body matches policy
+						errors.push(
+							`Expected body of script "${requiredScript.name}" to be "${
+								requiredScript.body
+							}". Found "${packageJson.scripts[requiredScript.name]}".`,
+						);
+					}
+				}
+			}
+
+			if (errors.length > 0) {
+				return [
+					`API-Extractor violations for public package "${packageJson.name}":`,
```

Comment:
These don't _have_ to be api-extractor specific, right?
```suggestion
					`Policy violations for public package "${packageJson.name}":`,
```

Reply:
Yes, good catch!